packages/amplify-graphql-model-transformer/src/resources/dynamo-model-resource-generator.ts

```diff
@@ -233,64 +233,70 @@ export class DynamoModelResourceGenerator extends ModelResourceGenerator {
   /**
    * createIAMRole
    */
-  createIAMRole = (context: TransformerContextProvider, def: ObjectTypeDefinitionNode, scope: Construct, tableName: string): iam.Role => {
+  createIAMRole = (context: TransformerContextProvider, def: ObjectTypeDefinitionNode, scope: Construct, tableName: string): iam.IRole => {
     const roleName = context.resourceHelper.generateIAMRoleName(ModelResourceIDs.ModelTableIAMRoleID(def!.name.value));
+    const amplifyDataStoreTableName = context.resourceHelper.generateTableName(SyncResourceIDs.syncTableName);
     const role = new iam.Role(scope, ModelResourceIDs.ModelTableIAMRoleID(def!.name.value), {
       roleName,
       assumedBy: new iam.ServicePrincipal('appsync.amazonaws.com'),
+      // Use an inline policy here to prevent unnecessary policy resources from being generated
+      // and slowing down deployments.
+      inlinePolicies: {
```

Comment:
Just FYI this also changes the default DynamoDB table in Gen1. We probably need to confirm this is not breaking change for them (test by deploying Gen1 in current CLI and update with this branch)

Reply:
That's right. From manual testing, this is not a breaking change. For a Gen1 app, a new `amplify push` will combine the two policies into one inline policy. No actions are added or removed from policy statements attached to this role.